samza-api/src/main/java/org/apache/samza/context/SamzaContainerContext.java

```diff
@@ -0,0 +1,17 @@
+package org.apache.samza.context;
+
+/**
+ * Container object for job and container context provided by the Samza framework to instantiate an application at
+ * runtime.
+ */
+public interface SamzaContainerContext {
```

Comment:
Why do we need this wrapper context? Looks like `Context` already provides JobContext and ContainerContext.

Reply:
Do you mean we need to pass this (and SamzaContext) around to users, or internally within the framework? Also, do we need both this and SamzaContext, or could we do with just one? Just one with both container and task level context might not be that bad, esp. if it's only used to create application defined context.

PS: Let's create a SEP that documents the class hierarchy. Will be useful for reference in this PR and for public documentation.

Reply:
Discussed offline. Instead of using SamzaContext and SamzaContainerContext, the individual context objects will be passed around when needed. This simplifies the object hierarchy/wiring. It is unlikely we will need to add a new type of context object, so the signatures should be able to stay the same.